release-22.1: changefeedccl: fix alter changefeed to handle targets with fully qualified names

sherman-grewal opened the pull request:
Backport 1/1 commits from #79542.

/cc @cockroachdb/release

---

changefeedccl: fix alter changefeed to handle targets with
fully qualified names

Resolves #79502

Currently, the ALTER CHANGEFEED statement would not work
with changefeeds that use fully qualified names in their
CREATE CHANGEFEED statement. This is because when we try
to include the existing targets, we would drop the
database/schema name that the user provided. Hence, when
we perform validation checks on the existing + new
changefeed targets, we would not be able to resolve the
existing targets that used fully qualified names and we
would return an error.

In this PR we ensure that we add each existing target
with their fully qualified name so that we can resolve
these targets in our validation checks. A consequence of
this change is that now every changefeed will display
the fully qualified name of every target in the
SHOW CHANGEFEED JOB query.

Release note (enterprise change): Fix the issue of not
being able to resolve existing targets when performing
validation checks by adding the fully qualified name
to the changefeed statement. As a result, every altered
changefeed will display the fully qualified name of every
target in the SHOW CHANGEFEED JOB query.

Release justification: Low risk bug fix.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79745)
<!-- Reviewable:end -->

Commit:
changefeedccl: fix alter changefeed to handle targets with
fully qualified names

Currently, the ALTER CHANGEFEED statement would not work
with changefeeds that use fully qualified names in their
CREATE CHANGEFEED statement. This is because when we try
to include the existing targets, we would drop the
database/schema name that the user provided. Hence, when
we perform validation checks on the existing + new
changefeed targets, we would not be able to resolve the
existing targets that used fully qualified names and we
would return an error.

In this PR we ensure that we add each existing target
with their fully qualified name so that we can resolve
these targets in our validation checks. A consequence of
this change is that now every changefeed will display
the fully qualified name of every target in the
SHOW CHANGEFEED JOB query.

Release note (enterprise change): Fix the issue of not
being able to resolve existing targets when performing
validation checks by adding the fully qualified name
to the changefeed statement. As a result, every altered
changefeed will display the fully qualified name of every
target in the SHOW CHANGEFEED JOB query.

Release justification: Low risk bug fix.
